### ISSUE TITLE
no matching function for call to 'Ticker::Ticker()'

### ISSUE THREAD

**Issue by melbis** [arendst/Sonoff-Tasmota#4499]
> Executing task in folder Tasmota: C:\Users\DimKas\.platformio\penv\Scripts\platformio.exe run <

Processing sonoff-classic (framework: arduino; platform: espressif8266@1.5.0; board: esp01_1m)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/esp01_1m.html
PLATFORM: Espressif 8266 > Espressif Generic ESP8266 ESP-01 1M
HARDWARE: ESP8266 80MHz 80KB RAM (1MB Flash)
Converting sonoff.ino
Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF MODES: FINDER(chain) COMPATIBILITY(soft)
Collected 51 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <LiquidCrystal_I2C>
|   |-- <Wire> 1.0
|-- <ESP8266WiFi> 1.0
|-- <Adafruit LED Backpack Library> 1.1.6
|   |-- <Adafruit GFX Library> 1.2.9
|   |   |-- <SPI> 1.0
|   |-- <Wire> 1.0
|-- <MQTT> 2.4.0
|-- <Ticker> 3.1.4
|-- <TasmotaSerial> 2.1.0
|-- <Adafruit ILI9341> 1.2.0
|   |-- <Adafruit GFX Library> 1.2.9
|   |   |-- <SPI> 1.0
|   |-- <SPI> 1.0
|-- <Adafruit CCS811 Library> 1.0.0
|   |-- <Wire> 1.0
|-- <Joba_Tsl2561> 2.0.7
|   |-- <Wire> 1.0
|-- <IRremoteESP8266> 2.2.1
|-- <ESP8266WebServer> 1.0
|   |-- <ESP8266WiFi> 1.0
|-- <TasmotaMqtt> 1.0.0
|-- <ArduinoOTA> 1.0
|   |-- <ESP8266mDNS>
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <ESP8266WiFi> 1.0
|-- <ArduinoJson> 5.11.2
|-- <OneWire> 2.3.2
|-- <rc-switch> 2.6.2
|-- <TasmotaModbus> 1.1.0
|   |-- <TasmotaSerial> 2.1.0
|-- <ESP KNX IP Library> 0.5.1
|   |-- <ESP8266WiFi> 1.0
|   |-- <EEPROM> 1.0
|   |-- <ESP8266WebServer> 1.0
|   |   |-- <ESP8266WiFi> 1.0
|-- <PubSubClient> 2.6
|-- <ESP8266HTTPClient> 1.1
|   |-- <ESP8266WiFi> 1.0
|-- <I2Cdevlib-MPU6050>
|   |-- <I2Cdevlib-Core>
|   |   |-- <Wire> 1.0
|-- <Adafruit GFX Library> 1.2.9
|   |-- <SPI> 1.0
|-- <Mutichannel_Gas_Sensor> 0.0.1
|   |-- <Wire> 1.0
|-- <Wire> 1.0
|-- <DNSServer> 1.1.0
|   |-- <ESP8266WiFi> 1.0
|-- <C2Programmer> 1.0.0
|-- <ESP8266mDNS>
|   |-- <ESP8266WiFi> 1.0
|-- <BME680_driver-bme680_v3.5.9>
|-- <SPI> 1.0
|-- <NeoPixelBus> 2.2.9
|   |-- <SPI> 1.0
|-- <ESP8266httpUpdate> 1.1
|   |-- <ESP8266HTTPClient> 1.1
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <ESP8266WiFi> 1.0
|-- <Adafruit SSD1306> 1.1.2
|   |-- <Adafruit GFX Library> 1.2.9
|   |   |-- <SPI> 1.0
|   |-- <Wire> 1.0
|   |-- <SPI> 1.0
|-- <I2Cdevlib-ADS1115>
|   |-- <I2Cdevlib-Core>
|   |   |-- <Wire> 1.0
|-- <Adafruit SGP30 Sensor> 1.0.2
|   |-- <Wire> 1.0
Compiling .pioenvs\sonoff-classic\src\sonoff.ino.cpp.o
Archiving .pioenvs\sonoff-classic\libe87\libTasmotaSerial-2.1.0.a
Archiving .pioenvs\sonoff-classic\libdb0\libAdafruit_ILI9341-1.2.0-Tasmota-1.0.a
Archiving .pioenvs\sonoff-classic\libfd9\libAdafruit_CCS811-1.0.0.14.a
Compiling .pioenvs\sonoff-classic\lib2fd\Joba_Tsl2561-2.0.7\Tsl2561.cpp.o
Compiling .pioenvs\sonoff-classic\lib2fd\Joba_Tsl2561-2.0.7\Tsl2561Util.cpp.o
Compiling .pioenvs\sonoff-classic\liba92\IRremoteESP8266-2.2.1.02\ir_Dish.cpp.o
Compiling .pioenvs\sonoff-classic\liba92\IRremoteESP8266-2.2.1.02\ir_Fujitsu.cpp.o
Compiling .pioenvs\sonoff-classic\liba92\IRremoteESP8266-2.2.1.02\ir_GlobalCache.cpp.o
Archiving .pioenvs\sonoff-classic\lib2fd\libJoba_Tsl2561-2.0.7.a
Compiling .pioenvs\sonoff-classic\liba92\IRremoteESP8266-2.2.1.02\ir_Gree.cpp.o
D:/IoT/VSCode/Tasmota/sonoff/support.ino:27:8: error: no matching function for call to 'Ticker::Ticker()'
Ticker tickerOSWatch;
^
D:/IoT/VSCode/Tasmota/sonoff/support.ino:27:8: note: candidates are:
In file included from D:/IoT/VSCode/Tasmota/sonoff/sonoff.ino:48:0:
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:61:2: note: Ticker::Ticker(fptr, uint32_t, uint16_t, resolution_t)
Ticker(fptr callback, uint32_t timer, uint16_t repeat = 0, resolution_t resolution = MICROS);
^
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:61:2: note:   candidate expects 4 arguments, 0 provided
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:49:7: note: constexpr Ticker::Ticker(const Ticker&)
class Ticker {
^
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:49:7: note:   candidate expects 1 argument, 0 provided
D:/IoT/VSCode/Tasmota/sonoff/support.ino: In function 'void OsWatchInit()':
D:/IoT/VSCode/Tasmota/sonoff/support.ino:65:17: error: 'class Ticker' has no member named 'attach_ms'
tickerOSWatch.attach_ms(((OSWATCH_RESET_TIME / 3) * 1000), OsWatchTicker);
^
D:/IoT/VSCode/Tasmota/sonoff/support.ino: At global scope:
D:/IoT/VSCode/Tasmota/sonoff/support.ino:1886:8: error: no matching function for call to 'Ticker::Ticker()'
Ticker TickerRtc;
^
D:/IoT/VSCode/Tasmota/sonoff/support.ino:1886:8: note: candidates are:
In file included from D:/IoT/VSCode/Tasmota/sonoff/sonoff.ino:48:0:
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:61:2: note: Ticker::Ticker(fptr, uint32_t, uint16_t, resolution_t)
Ticker(fptr callback, uint32_t timer, uint16_t repeat = 0, resolution_t resolution = MICROS);
^
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:61:2: note:   candidate expects 4 arguments, 0 provided
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:49:7: note: constexpr Ticker::Ticker(const Ticker&)
class Ticker {
^
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:49:7: note:   candidate expects 1 argument, 0 provided
D:/IoT/VSCode/Tasmota/sonoff/support.ino: In function 'void RtcInit()':
D:/IoT/VSCode/Tasmota/sonoff/support.ino:2253:13: error: 'class Ticker' has no member named 'attach'
TickerRtc.attach(1, RtcSecond);
^
D:/IoT/VSCode/Tasmota/sonoff/xdrv_03_energy.ino: At global scope:
D:/IoT/VSCode/Tasmota/sonoff/xdrv_03_energy.ino:82:8: error: no matching function for call to 'Ticker::Ticker()'
Ticker ticker_energy;
^
D:/IoT/VSCode/Tasmota/sonoff/xdrv_03_energy.ino:82:8: note: candidates are:
In file included from D:/IoT/VSCode/Tasmota/sonoff/sonoff.ino:48:0:
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:61:2: note: Ticker::Ticker(fptr, uint32_t, uint16_t, resolution_t)
Ticker(fptr callback, uint32_t timer, uint16_t repeat = 0, resolution_t resolution = MICROS);
^
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:61:2: note:   candidate expects 4 arguments, 0 provided
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:49:7: note: constexpr Ticker::Ticker(const Ticker&)
class Ticker {
^
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:49:7: note:   candidate expects 1 argument, 0 provided
D:/IoT/VSCode/Tasmota/sonoff/xdrv_03_energy.ino: In function 'void EnergySnsInit()':
D:/IoT/VSCode/Tasmota/sonoff/xdrv_03_energy.ino:521:19: error: 'class Ticker' has no member named 'attach_ms'
ticker_energy.attach_ms(200, Energy200ms);
^
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino: At global scope:
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino:32:8: error: no matching function for call to 'Ticker::Ticker()'
Ticker TickerMSearch;
^
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino:32:8: note: candidates are:
In file included from D:/IoT/VSCode/Tasmota/sonoff/sonoff.ino:48:0:
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:61:2: note: Ticker::Ticker(fptr, uint32_t, uint16_t, resolution_t)
Ticker(fptr callback, uint32_t timer, uint16_t repeat = 0, resolution_t resolution = MICROS);
^
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:61:2: note:   candidate expects 4 arguments, 0 provided
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:49:7: note: constexpr Ticker::Ticker(const Ticker&)
class Ticker {
^
C:\users\dimkas\.platformio\lib\Ticker_ID1586/Ticker.h:49:7: note:   candidate expects 1 argument, 0 provided
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino: In function 'void WemoRespondToMSearch(int)':
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino:82:17: error: 'class Ticker' has no member named 'detach'
TickerMSearch.detach();
^
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino: In function 'void HueRespondToMSearch()':
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino:161:17: error: 'class Ticker' has no member named 'detach'
TickerMSearch.detach();
^
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino: In function 'void PollUdp()':
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino:250:27: error: 'class Ticker' has no member named 'attach_ms'
TickerMSearch.attach_ms(UDP_MSEARCH_SEND_DELAY, WemoRespondToMSearch, 1);
^
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino:256:27: error: 'class Ticker' has no member named 'attach_ms'
TickerMSearch.attach_ms(UDP_MSEARCH_SEND_DELAY, WemoRespondToMSearch, 2);
^
D:/IoT/VSCode/Tasmota/sonoff/xplg_wemohue.ino:265:27: error: 'class Ticker' has no member named 'attach_ms'
TickerMSearch.attach_ms(UDP_MSEARCH_SEND_DELAY, HueRespondToMSearch);
^
Compiling .pioenvs\sonoff-classic\liba92\IRremoteESP8266-2.2.1.02\ir_JVC.cpp.o
Compiling .pioenvs\sonoff-classic\liba92\IRremoteESP8266-2.2.1.02\ir_Kelvinator.cpp.o
*** [.pioenvs\sonoff-classic\src\sonoff.ino.cpp.o] Error 1
=============================================================================================== [ERROR] Took 4.56 seconds ===============================================================================================

======================================================================================================= [SUMMARY] =======================================================================================================
Environment sonoff              [SKIP]
Environment sonoff-minimal      [SKIP]
Environment sonoff-basic        [SKIP]
Environment sonoff-classic      [ERROR]
Environment sonoff-knx          [SKIP]
Environment sonoff-sensors      [SKIP]
Environment sonoff-display      [SKIP]
Environment sonoff-BG           [SKIP]
Environment sonoff-BR           [SKIP]
Environment sonoff-CN           [SKIP]
Environment sonoff-CZ           [SKIP]
Environment sonoff-DE           [SKIP]
Environment sonoff-ES           [SKIP]
Environment sonoff-FR           [SKIP]
Environment sonoff-GR           [SKIP]
Environment sonoff-HE           [SKIP]
Environment sonoff-HU           [SKIP]
Environment sonoff-IT           [SKIP]
Environment sonoff-NL           [SKIP]
Environment sonoff-PL           [SKIP]
Environment sonoff-PT           [SKIP]
Environment sonoff-RU           [SKIP]
Environment sonoff-TR           [SKIP]
Environment sonoff-TW           [SKIP]
Environment sonoff-UK           [SKIP]
=============================================================================================== [ERROR] Took 4.58 seconds ===============================================================================================
The terminal process terminated with exit code: 1

Terminal will be reused by tasks, press any key to close it.

**Comment by ascillato2** [comment]
Hi,

Tasmota is compiling fine. There is no issue in the codebase, so that means that you have not a complete copy of Tasmota or an issue with your directories. Please Delete All and download the code again. 

Thanks.

If you need further assistance, please go to the Tasmota support chat. Thanks.

### Support Information
See [Wiki](https://github.com/arendst/Sonoff-Tasmota/wiki) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.
